Fix Prim's algorithm implementation and update tests

Alokih opened the pull request:
Fixes: #28 

**Description:**

This PR fixes the implementation of Prim's algorithm to align with the API conventions.
- Removed explicit return in the `run` method.
- Added `reconstructMST` method to retrieve the edges.
- Updated tests to use `reconstructMST` instead of relying on the return value of `run`.

Ducasse commented:
Thanks for the PR. What is reconstructMST?

Alokih commented:
Hi @Ducasse, I think I made a mistake in naming the method here, In prim's algo MST edges are already computed and stored, so there’s no need to "reconstruct" them. 

Do you think I should name it as something like getMSTEdges ( Here MST stands for Minimum Spanning Tree ).
Please let me know if you have some other suggestions.

Ducasse commented:
I would call them minimumSpanningTreeEdges (get feels like a Java hack).


Now I do not get the name of the variable because does it returns a tree or edges?

```
prim run.
tree := prim reconstructMST collect: [ :e | e asTuple ].
```

Alokih commented:
You're right method returns collection of edges & not a tree, could you please suggest what should I rename variable to.

Does 'treeEdges' make sense ?

Alokih commented:
Thnx @Ducasse, I have renamed the variables.

Please have a look whenever you get time.

jordanmontt commented:
Nice refactor :)
the name `minimumSpanningTreeEdges` it is way better.

Thanks for your PR @Alokih

jordanmontt commented:
BTW we have a failing CI (it is not your fault). I believe it can be a good exercise for you if you can make a PR to have all the CIs green. Of course, only if it interest you. What do you think @Alokih ?

Alokih commented:
For sure @jordanmontt I am working on it. Thnx

Alokih commented:
@jordanmontt I tested the workflow run on my fork repo & found 2 main issues :

1. There was a mismatch in naming of `AINetworkFlowEdge.class.st`. Actual file was named `AINetworkFLowEdge.class.st` in repo ( Note the uppercase L in FLow ).
So I fixed it & again ran workflow then I got another issue 
Here is the workflow run link in forked repo [link](https://github.com/Alokih/graph-algorithms/actions/runs/13758684725)

2. CI was failing for `Pharo64-9.0` specifically on the Windows runner, the issue was related to VM I think ( `sqMakeMemoryNotExecutableFromTo` in `sqWin32SpurAlloc.c` ) You could also check from above workflow link I provided.
So what I did was removed `Pharo64-9.0` from jobs & ran again 
Here is the [link](https://github.com/Alokih/graph-algorithms/actions/runs/13759381088)
Please let me know what you think, if `Pharo64-9.0` is not crictical we could remove it either only for windows or all OS.
If it is required can you please give me some suggestions here, I am stuck at this point.

Also I have fixed this warning 
![Screenshot from 2025-03-10 14-22-25](https://github.com/user-attachments/assets/dccaf1b7-1a9f-4d6d-a2a1-7f5926790838)

jordanmontt commented:
Hello @Alokih 

1. Yes, you are right about the uppercase L. I don't know how it got there... Somebody must have edited the file manually.
2. Indeed, there is a VM failure for the Pharo 9 on WINDOWS. It is working just fine on my mac. What I just did it is just to remove Pharo 9, if we have users that need it in Pharo 9 we will investigate further.

I fixed those 2 things and know we have a green CI. Thanks for your investigation!!

Alokih commented:
Glad I could help ! @jordanmontt  Happy to see CI green now :tada:
